form/unit_testing.rst

```diff
@@ -182,31 +182,22 @@ allows you to return a list of extensions to register::
     namespace Tests\AppBundle\Form\Type;
 
     // ...
-    use AppBundle\Form\Type\TestedType;
     use Symfony\Component\Form\Extension\Validator\ValidatorExtension;
-    use Symfony\Component\Form\Form;
-    use Symfony\Component\Validator\ConstraintViolationList;
-    use Symfony\Component\Validator\Mapping\ClassMetadata;
-    use Symfony\Component\Validator\Validator\ValidatorInterface;
+    use Symfony\Component\Validator\Validation;
 
     class TestedTypeTest extends TypeTestCase
     {
-        private $validator;
-
         protected function getExtensions()
         {
-            $this->validator = $this->createMock(ValidatorInterface::class);
-            // use getMock() on PHPUnit 5.3 or below
-            // $this->validator = $this->getMock(ValidatorInterface::class);
-            $this->validator
-                ->method('validate')
-                ->will($this->returnValue(new ConstraintViolationList()));
-            $this->validator
-                ->method('getMetadataFor')
-                ->will($this->returnValue(new ClassMetadata(Form::class)));
+            $validator = Validation::createValidator();
```

Comment:
I would keep the current private property and do:
```php
if (null === $this->validator) {
    $this->validator = Validation::createValidator();
    // ...
}
```
Otherwise the `setUp` will be heavy , so also maybe a comment or an implementation of `tearDownAfterClass`.

Reply:
I don't really see for what you would need to access the validator. Previously this was just the case because you had to define in each tests individually which constraint violations should be raised.

Reply:
The property would not be meant to access it anymore but to prevent creating the validator for running each test in a class, performances wise.
We just need to create it once now that we don't need to control the mock anymore.

Reply:
That makes sense but is it really worth it? Wouldn't it just make the code harder to understand compared to the provided benefit?

Reply:
What about a PR to see the diff? You could advise to have that a logic in a trait. We could even provide such trait in core.